src/test/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/OffenderProgramProfileRepositoryTest.java

```diff
@@ -90,6 +94,119 @@ void getActivitiesByBookingIdAndProgramStatus_ReturnsNothing() {
 
         assertThat(activities).isEmpty();
     }
+
+    @Test
+    void findByNomisIdAndProgramStatusAndEndDateAfter() {
+        final var activities = repository.findByNomisIdAndProgramStatusAndEndDateAfter("A1234AC", List.of("ALLOC", "END"), LocalDate.of(2021, 1, 1)).stream()
+            .sorted((o1, o2) -> (int) (o2.getOffenderProgramReferenceId() - o1.getOffenderProgramReferenceId()))
+            .collect(Collectors.toList());
+
+        assertThat(activities).usingElementComparatorIgnoringFields("offenderProgramReferenceId",
```

Comment:
Matter of opinion again but using assertThat(activities).extracting().contains() might cut down the amount of noise here.